src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoroelasticKernel.hpp

```diff
@@ -0,0 +1,444 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SinglePhasePoroelasticKernel.hpp
+ */
+
+#ifndef GEOSX_PHYSICSSOLVERS_MULTIPHYSICS_SINGLEPHASEPOROELASTICKERNEL_HPP_
+#define GEOSX_PHYSICSSOLVERS_MULTIPHYSICS_SINGLEPHASEPOROELASTICKERNEL_HPP_
+#include "finiteElement/kernelInterface/ImplicitKernelBase.hpp"
+#include "physicsSolvers/fluidFlow/SinglePhaseBase.hpp"
+
+
+namespace geosx
+{
+
+namespace PoroelasticKernels
+{
+
+/**
+ * @brief Implements kernels for solving quasi-static equilibrium.
+ * @copydoc geosx::finiteElement::ImplicitKernelBase
+ * @tparam NUM_NODES_PER_ELEM The number of nodes per element for the
+ *                            @p SUBREGION_TYPE.
+ * @tparam UNUSED An unused parameter since we are assuming that the test and
+ *                trial space have the same number of support points.
+ *
+ * ### QuasiStatic Description
+ * Implements the KernelBase interface functions required for solving the
+ * quasi-static equilibrium equations using one of the
+ * "finite element kernel application" functions such as
+ * geosx::finiteElement::RegionBasedKernelApplication.
+ *
```

Comment:
idem to solid elastic solvers?

Reply:
I briefly modified the description, but more documentation for these kernels will be added. Note that the pressure interpolation (along with the global component densities for the multiphase case) is implicitly assumed piecewise constant since our flow solvers are either based on `FMV` or `HybridFVM`. A formal description of the functional space used to approximate all primary variable is needed.